apps/backend/src/services/userServices.ts

```diff
@@ -244,39 +263,76 @@ const updateUser = async (body: unknown, userId: number): Promise<UserWithoutPas
     error.name = 'InvalidDateError';
     throw error;
   }
-  let updateData: UpdateUserData | UpdateUserInput;
+
+  if (body.image_url) {
+    const imagePath = await uploadImage(body.image_url as string, `profile-pics/${userId}`);
+    if (typeof imagePath !== 'string') {
+      const error = new Error(imagePath.error);
+      error.name = 'InvalidInputError';
+      throw error;
+    }
+
+    body.image_url = imagePath;
+  }
+
+  let updateData: UpdateUserData;
 
   if (body.dob && typeof body.dob === 'string') {
     updateData = {
-      ...body, dob: new Date(body.dob)
-    }
+      ...body,
+      dob: new Date(body.dob),
+    } as UpdateUserData;
   } else {
-    updateData = body;
+    updateData = { ...body, dob: undefined };
   }
 
-  const updatedProfile: UserWithoutPasswordHash = await prismaClient.user.update({
-    where: { id: userId },
-    data: updateData,
-    // Ensures password hash isn't being sent
-    select: {
-      id: true,
-      username: true,
-      email: true,
-      first_name: true,
-      last_name: true,
-      dob: true,
-      gender_id: true,
-      bio: true,
+  try {
+    const updatedProfile: UserWithoutPasswordHash = await prismaClient.user.update({
+      where: { id: userId },
+      data: updateData,
+      // Ensures password hash isn't being sent
+      select: {
+        id: true,
+        username: true,
+        email: true,
+        first_name: true,
+        last_name: true,
+        dob: true,
+        gender_id: true,
+        bio: true,
+        image_url: true,
+      },
+    });
+
+    const subscriberInfo = {
+      subscriberId: String(updatedProfile.id),
+      firstName: updatedProfile.first_name || '',
+      lastName: updatedProfile.last_name || '',
+      email: updatedProfile.email,
+      avatar: updatedProfile.image_url ? URL + updatedProfile.image_url : '',
```

Comment:
Is this supposed to show the avatar of the user who triggered the notification? If that is so, then it's not working unfortunately.

Reply:
It doesn't work, but I think it's an issue with the novu API. Haven't figured it out and they haven't got back to me with a working answer.